lib/msal-common/src/client/AuthorizationCodeClient.ts

```diff
@@ -248,21 +248,26 @@ export class AuthorizationCodeClient extends BaseClient {
         }
 
         // Add sid or loginHint with preference for sid -> loginHint -> username of AccountInfo object
-        if (request.sid) {
-            parameterBuilder.addSid(request.sid);
-        } else if (request.account) {
-            const accountSid = this.extractAccountSid(request.account);
-            // If account and loginHint are provided, we will check account first for sid before adding loginHint
-            if (accountSid) {
-                parameterBuilder.addSid(accountSid);
+        if (request.prompt !== PromptValue.SELECT_ACCOUNT) {
```

Comment:
Would we want to also ignore account hints for login?

Reply:
I went back and forth on that. In the `prompt=login` scenario `login_hint` could be used to pre-fill the email field so maybe just ignore `sid`?

Reply:
I think I've seen scenarios where an app would want to trigger the login prompt with the current user, so as long as we provide a way to set prompt=login with and without an account hint, we should be good.

Reply:
And yeah, sid can be ignored for interactive requests.